keras_cv/models/segmentation/deeplab.py

```diff
@@ -56,136 +71,102 @@ class DeepLabV3(tf.keras.models.Model):
-
-    def train_step(self, data):
-        images, y_true, sample_weight = tf.keras.utils.unpack_x_y_sample_weight(data)
-        with tf.GradientTape() as tape:
-            y_pred = self(images, training=True)
-            total_loss = self.compute_loss(images, y_true, y_pred, sample_weight)
-            reg_losses = []
-            if self.weight_decay:
-                for var in self.trainable_variables:
-                    if "bn" not in var.name:
-                        reg_losses.append(self.weight_decay * tf.nn.l2_loss(var))
-                l2_loss = tf.math.add_n(reg_losses)
-                total_loss += l2_loss
-        self.optimizer.minimize(total_loss, self.trainable_variables, tape=tape)
-        return self.compute_metrics(images, y_true, y_pred, sample_weight=sample_weight)
-
-    def get_config(self):
-        config = {
-            "classes": self.classes,
-            "backbone": self._backbone_passed,
-            "decoder": self._decoder_passed,
-            "segmentation_head": self._segmentation_head_passed,
-        }
-        base_config = super().get_config()
-        return dict(list(base_config.items()) + list(config.items()))
+    inputs = utils.parse_model_inputs(input_shape, input_tensor)
+    height = input_shape[0]
+    width = input_shape[1]
+
+    x = inputs
+
+    if include_rescaling:
+        x = layers.Rescaling(1 / 255.0)(x)
+
+    if isinstance(backbone, str):
+        supported_premade_backbone = [
+            "resnet50_v2",
+        ]
+        if backbone not in supported_premade_backbone:
+            raise ValueError(
+                "Supported premade backbones are: "
+                f'{supported_premade_backbone}, received "{backbone}"'
+            )
+
+        if backbone == "resnet50_v2":
+            backbone = get_resnet_backbone(
+                backbone_weights, include_rescaling, **kwargs
+            )
+
+    else:
+        # TODO(scottzhu): Might need to do more assertion about the model
+        # What else do we want to test for? Shapes? This feels like too little, but
+        # more assertions feel like they'd be limiting.
+        if not isinstance(backbone, tf.keras.layers.Layer):
+            raise ValueError(
+                "Backbone need to be a `tf.keras.layers.Layer`, " f"received {backbone}"
+            )
+
+    feature_map = backbone(x)
+    output = SpatialPyramidPooling(dilation_rates=[6, 12, 18])(feature_map)
+    output = tf.keras.layers.UpSampling2D(
+        size=(height // feature_map.shape[1], width // feature_map.shape[2]),
+        interpolation="bilinear",
+    )(output)
+
+    if segmentation_head is None:
+        segmentation_head = get_segmentation_head(classes)
+
+    output = segmentation_head(output)
+    model = tf.keras.Model(inputs, output, name=name, **kwargs)
+
+    if backbone_weights is not None:
+        backbone.load_weights(backbone_weights)
+
+    return model
+
+
+def get_segmentation_head(classes):
+    return tf.keras.Sequential(
```

Comment:
Can you set name for the segmentation model? It will be convinient to visualize with `plot_model` function. ex,

```python
Sequential(
    layers=None, name='segmentation head'
)
```

Reply:
Exchanged the Sequential seg head for the modular component, which is a single layer now.
I'll add the name to the layer

Reply:
<img width="566" alt="image" src="https://user-images.githubusercontent.com/60978046/206963258-9b5c9de5-0ab3-41fc-9ad2-a986f579aa8b.png">